R/g.sib.sum.R

```diff
@@ -17,16 +17,31 @@ g.sib.sum = function(SLE,M,ignorenonwear=TRUE,desiredtz="") {
   colnames(sleep) = colnames(A)[(which(colnames(A)=="night")+1):ncol(A)]
   ws3 = M$windowsizes[1]
   ws2 = M$windowsizes[2]
-  sib.cla.sum = as.data.frame(matrix(0, 1000,9))
+  sib.cla.sum = as.data.frame(matrix(-1, 1000,9))
+  # label night after last night as -1 to differentiate it from night 0
+  lastNightEnd = max(which(night == max(night)))
+  if (length(night) > lastNightEnd) {
+    night[(lastNightEnd + 1):length(night)] = -1
+  }
+  # unique nights
   un = unique(night) #unique nights
-  missingnights = which(un == 0 | is.na(un) == TRUE)
```

Comment:
what was the meaning of `un == 0`? do we not need it anymore?

Reply:
`un` means unique nights, and it is a vector that later is named `nnights.list`. For example, `un = c(0, 1, 2, 3)` means that there is an incomplete night at the beginning (starting after midnight) and 2 complete nights later on (incomplete = does not include midnight timestamp). 

In this branch, the last incomplete night (in case the recording finishes after noon) is recoded to `-1`, and this one is always removed from the `nnights.list`. Below in this function, the first incomplete night is also removed if recording starts after 4am (line 36)